privacyidea/lib/smtpserver.py

```diff
@@ -56,17 +59,18 @@ def __init__(self, db_smtpserver_object):
 
     def send_email(self, recipient, subject, body, sender=None,
                    reply_to=None, mimetype="plain"):
-        return self.test_email(self.config, recipient, subject, body, sender,
-                               reply_to, mimetype)
+        return send_or_enqueue_email(self.config.get(), recipient, subject, body, sender,
+                                     reply_to, mimetype)
 
     @staticmethod
+    @job(SEND_EMAIL_JOB_NAME, fire_and_forget=True)
     def test_email(config, recipient, subject, body, sender=None,
                    reply_to=None, mimetype="plain"):
         """
-        Sends an email via the SMTP Database Object
+        Sends an email via the configuration.
 
         :param config: The email configuration
-        :type config: SMTPServer Database Model
+        :type config: dict
```

Comment:
Assume the privacyIDEA process wants to send a "smtpserver.send_email" job to the queue (like "send mail to foo@example.com"). For that, it needs to store the arguments in the Redis database, so the arguments need to be serialized in some way. I thought it would be easier to serialize a dictionary instead of a ``SMTPServer`` object. But I'm currently not really sure how the objects are serialized -- I'll check that!

Reply:
I checked: It *is* possible to use a ``SMTPServer`` database model here. huey uses pickle to serialize the arguments, and SQLAlchemy supports serializing model objects. However, using a dictionary seems cleaner to me, and it also reduces the size of the serialized data: With a ``SMTPServer`` object, ~1000 bytes are written to the Redis store. With a dictionary, only ~500 bytes are written to the Redis store.

What do you think?